### DISCUSSION TITLE
lom204

### DISCUSSION THREAD

**Discussion by tcpipchip** [Lora-net/LoRaMac-node#1231] [Q&A]
I ported the LoRaMac-node to the LOM204 (WISOL STM32L071+SX1276)

![image](https://user-images.githubusercontent.com/11488460/148060567-e9e7362b-a42b-401f-9fe7-d40d4cf56af7.png)

But i have a big question, TXD and RXD uses PA2 and PA3, if you look the **board-config.h**

```
#define UART_TX                                     PA_2
#define UART_RX                                     PA_3
```
i changed to PA_2 to PA_9, compiles, burn, but nothing echos on PA_9 pin of LOM204.

Something else that i have to modify ?

Thanks!

**Comment by MedadRufus** [comment]
Have a look to see if your new board uses Uart1 or Uart2. There could be a mismatch.

**Reply by tcpipchip** [reply]
On keil c i use UART1 and UART5 and works fine
On arduino c i use too UART1 and UART5 and works nice too

**Reply by MedadRufus** [reply]
In the loramac-node code if you will have to ensure the uart port has been set. You may have to change it to uart1 or uart5.
https://github.com/Lora-net/LoRaMac-node/blob/edff6ce161773e888687a9cd352ce2e2234e6b0d/src/boards/B-L072Z-LRWAN1/board.c#L127-L128

**Comment by tcpipchip** [comment]
Do you mean change UART_2 to UART_5 and UART_TX to PA_9 and UART_RX to PA_10 ?

btw
Here’s a beta blog
https://ttn-gateway-lora.blogspot.com/2022/01/wisol-lom204a02-acessando-ttn-v3-otaa.html?m=1

**Reply by MedadRufus** [reply]
You may have to change UART_2 to UART_1. UART_5 is not defined(it can be defined by you as necessary) and since you have UART_1 accessible, you might as well use it.

Looking at your blog, it looks like the code is from a long ago release of the LoRaMAC node code, and so I would recommend moving to using the latest code on this repo.
